### PR TITLE
feat: add support for popup timeout

### PR DESCRIPTION
**Pull request by douglaseggleton**
# Description

Adds support for cancelling popup after x seconds.  This also emits the reason for cancellation - timeout, cancel (from the merchant overlay), or close (closing the popup). We can use this to display an appropriate error message in the UI.

Fixes # (issue)

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own changes
- [ ] I have run `yarn lint` to make sure my changes pass all tests
- [ ] I have run `yarn test` to make sure my changes pass all linters
- [ ] I have pulled the latest changes from the upstream `main` branch
- [ ] I have tested both the react and the CDN versions on local and integration environments
- [ ] I have added the necessary labels to this PR in case a new release needs to be published after merging into `main` (e.g. `release` and `patch`)

## Contribution guidelines

For contribution guidelines, styleguide, and other helpful information please
see the `CONTRIBUTING.md` file in the root of this project.

### CODE REVIEW

File: packages/embed/src/index.ts [hunk 1]
```diff
@@ -114,7 +114,8 @@ export function setup(setupConfig: SetupConfig): EmbedInstance {
 
   createPopupController(
     mutableRef<{ popup: Window; stopCallback: () => void }>(),
-    subjectManager
+    subjectManager,
+    config.popupTimeout ? config.popupTimeout * 60 : null
```
Comment (luca-gr4vy) [inline packages/embed/src/index.ts:118]: Shouldn't this be converted to milliseconds (`config.popupTimeout * 1000`)?

Comment (douglaseggleton) [reply packages/embed/src/index.ts:118]: 🤦🏻‍♂️ Thanks!

Comment (douglaseggleton) [inline packages/embed/src/index.ts:118]:
```suggestion
    config.popupTimeout ? config.popupTimeout * 1000 : null
```

**Comment by douglaseggleton** [comment]
Closing in favour of https://github.com/gr4vy/gr4vy-embed/pull/115

**Pull request closed without merging**